build/php.m4

```diff
@@ -2735,3 +2735,17 @@ AC_DEFUN([PHP_PATCH_CONFIG_HEADERS], [
   $SED -e 's/^#undef PACKAGE_[^ ]*/\/\* & \*\//g' < $srcdir/$1 \
     > $srcdir/$1.tmp && mv $srcdir/$1.tmp $srcdir/$1
 ])
+
+dnl
+dnl PHP_REMOVE_OPTIMIZATION_FLAGS
+dnl
+dnl Removes known compiler optimization flags like -O, -O0, -O1, ..., -Ofast
+dnl from CFLAGS and CXXFLAGS.
+dnl
+AC_DEFUN([PHP_REMOVE_OPTIMIZATION_FLAGS], [
+  changequote({,})
+  sed_script='s/\([\t ]\|^\)-O\([0-9gsz]\|fast\|\)\([\t ]\|$\)/\1/g'
+  changequote([,])
+  CFLAGS=`echo "$CFLAGS" | sed -e "$sed_script"`
+  CXXFLAGS=`echo "$CFLAGS" | sed -e "$sed_script"`
```

Comment:
```suggestion
  sed_script='s/\([[\t ]]\|^\)-O\([[0-9gsz]]\|fast\|\)\([[\t ]]\|$\)/\1/g'
  CFLAGS=$(echo "$CFLAGS" | $SED -e "$sed_script")
  CXXFLAGS=$(echo "$CFLAGS" | $SED -e "$sed_script")
```

Reply:
Ah, sorry and another one. Probably the most important one as CXXFLAGS should be its own variable instead. 

```diff
diff --git a/build/php.m4 b/build/php.m4
index dee47f6e1a..e13fc3367e 100644
--- a/build/php.m4
+++ b/build/php.m4
@@ -2745,5 +2745,5 @@ dnl
 AC_DEFUN([PHP_REMOVE_OPTIMIZATION_FLAGS], [
   sed_script='s/\([[\t ]]\|^\)-O\([[0-9gsz]]\|fast\|\)\([[\t ]]\|$\)/\1/g'
   CFLAGS=$(echo "$CFLAGS" | $SED -e "$sed_script")
-  CXXFLAGS=$(echo "$CFLAGS" | $SED -e "$sed_script")
+  CXXFLAGS=$(echo "$CXXFLAGS" | $SED -e "$sed_script")
 ])
```

Reply:
Nice catch!

Reply:
Thank you!